lib/carto/authentication_manager.rb

```diff
@@ -0,0 +1,26 @@
+module Carto
+  class AuthenticationManager
+
+    def self.validate_session(warden_context, request, user, reset_session_on_error = true)
+      if session_security_token_valid?(warden_context, user)
+        true
+      else
+        request.reset_session if reset_session_on_error
+        false
+      end
```

Comment:
Minor styling suggestion to avoid conditionals for readability
```suggestion
      return true if session_security_token_valid?(warden_context, user)
      request.reset_session if reset_session_on_error
      return false
```